setup.py

```diff
@@ -0,0 +1,95 @@
+from setuptools import setup
+import os
+import sys
+import warnings
+
+# Loosely based on NILMTK's setup.py
+
+TRAVIS_TAG = os.environ.get('TRAVIS_TAG', '')
+
+if TRAVIS_TAG:
+    #TODO: validate if the tag is a valid version number
+    VERSION = TRAVIS_TAG
+    ISRELEASED = not ('dev' in TRAVIS_TAG)
+    QUALIFIER = ''
+else:
+    MAJOR = 0
+    MINOR = 1
+    MICRO = 0
+    DEV = 1 # For multiple dev pre-releases, please increment this value
+    ISRELEASED = False
+    VERSION = '%d.%d.%d' % (MAJOR, MINOR, MICRO)
+    QUALIFIER = ''
+
+
+FULLVERSION = VERSION
+if not ISRELEASED and not TRAVIS_TAG:
+    try:
+        import subprocess
+        try:
+            pipe = subprocess.Popen(["git", "rev-parse", "--short", "HEAD"],
+                                    stdout=subprocess.PIPE).stdout
+        except OSError:
+            # msysgit compatibility
+            pipe = subprocess.Popen(
+                ["git.cmd", "rev-parse", "--short", "HEAD"],
+                stdout=subprocess.PIPE).stdout
+        rev = pipe.read().strip()
+        # makes distutils blow up on Python 2.7
+        if sys.version_info[0] >= 3:
+            rev = rev.decode('ascii')
+
+        # Use a local version tag to include the git revision
+        FULLVERSION += ".dev{}+git.{}".format(DEV, rev)
+    except:
+        FULLVERSION += ".dev{}".format(DEV)
+        warnings.warn('WARNING: Could not get the git revision, version will be "{}"'.format(FULLVERSION))
+else:
+    FULLVERSION += QUALIFIER
+
+
+def write_version_py(filename=None):
+    cnt = """\
+version = '%s'
+short_version = '%s'
+"""
+    if not filename:
+        filename = os.path.join(
+            os.path.dirname(__file__), 'nilmtk_contrib', 'version.py')
```

Comment:
The `setup.py` fails at this point on my machine. Reason is that the `nilmtk_contrib` directory does not exist inside the repository.
The current repository path for the python package is `./disaggregate/` but should be `./nilmtk_contrib/disaggregate/`.

Reply:
It's not supposed to be used with the master file structure as-is. Check the other changes from the PR commits, files are moved/renamed too.

Reply:
My bad, your file rename is correct.